qiskit/visualization/gate_map.py

```diff
@@ -705,24 +908,78 @@ def plot_error_map(backend, figsize=(12, 9), show_title=True):
-
-    num_qubits = config["n_qubits"]
-
-    # sx error rates
-    single_gate_errors = [0] * num_qubits
-    for gate in props["gates"]:
-        if gate["gate"] == "sx":
-            _qubit = gate["qubits"][0]
-            for param in gate["parameters"]:
-                if param["name"] == "gate_error":
-                    single_gate_errors[_qubit] = param["value"]
-                    break
-            else:
-                raise VisualizationError(
-                    f"Backend '{backend}' did not supply an error for the 'sx' gate."
-                )
+    backend_version = _get_backend_interface_version(backend)
+    if backend_version <= 1:
+        num_qubits = backend.configuration().n_qubits
+        cmap = backend.configuration().coupling_map
+        props = backend.properties().to_dict()
+        single_gate_errors = [0] * num_qubits
+        read_err = [0] * num_qubits
+        cx_errors = []
+        # sx error rates
+        for gate in props["gates"]:
+            if gate["gate"] == "sx":
+                _qubit = gate["qubits"][0]
+                for param in gate["parameters"]:
+                    if param["name"] == "gate_error":
+                        single_gate_errors[_qubit] = param["value"]
+                        break
+                else:
+                    raise VisualizationError(
+                        f"Backend '{backend}' did not supply an error for the 'sx' gate."
+                    )
+        if cmap:
+            directed = False
+            if num_qubits < 20:
+                for edge in cmap:
+                    if not [edge[1], edge[0]] in cmap:
+                        directed = True
+                        break
+
+            for line in cmap:
+                for item in props["gates"]:
+                    if item["qubits"] == line:
+                        cx_errors.append(item["parameters"][0]["value"])
+                        break
+                else:
+                    continue
+        for qubit in range(num_qubits):
+            for item in props["qubits"][qubit]:
+                if item["name"] == "readout_error":
+                    read_err.append(item["value"])
+
+    else:
+        num_qubits = backend.num_qubits
+        cmap = backend.coupling_map
+        two_q_error_map = {}
+        single_gate_errors = [0] * num_qubits
+        read_err = [0] * num_qubits
+        cx_errors = []
+        for gate, prop_dict in backend.target.items():
+            if prop_dict is None or None in prop_dict:
+                continue
+            for qargs, inst_props in prop_dict.items():
+                if gate == "measure":
+                    if inst_props.error is not None:
+                        read_err[qargs[0]] = inst_props.error
+                elif len(qargs) == 1:
+                    if inst_props.error is not None:
+                        single_gate_errors = max(single_gate_errors[qargs[0]], inst_props.error)
```

Comment:
```suggestion
                        single_gate_errors[qargs[0]] = max(single_gate_errors[qargs[0]], inst_props.error)
```

Should it be this instead?

Reply:
Yep good call, as you can tell there isn't great test coverage on this code. It's something we should fix I think there is an issue related to this: https://github.com/Qiskit/qiskit-terra/issues/7691 although we should expand the coverage as part of this.